src/game/network/framemetrics.cpp

```diff
@@ -0,0 +1,114 @@
+/**
+ * @file
+ *
+ * @author tomsons26
+ *
+ * @brief
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include "framemetrics.h"
+#include "display.h"
+#include "globaldata.h"
+#include "networkutil.h"
+#include "rtsutils.h"
+
+FrameMetrics::FrameMetrics() :
+    m_averageFps(0.0f),
+    m_averageLatency(0.0f),
+    m_cushionIndex(0),
+    m_fpsListIndex(0),
+    m_lastFpsTimeThing(0),
+    m_minimumCushion(0)
+{
+    m_pendingLatencies = new int[MAX_FRAMES_AHEAD];
+
+    for (int i = 0; i < MAX_FRAMES_AHEAD; ++i) {
+        m_pendingLatencies[i] = 0;
+    }
+
+    m_fpsList = new float[g_theWriteableGlobalData->m_networkFPSHistoryLength];
+
+    m_latencyList = new float[g_theWriteableGlobalData->m_networkLatencyHistoryLength];
```

Comment:
`m_fpsList` and `m_latencyList` are not default initialized here. It seems to be no practical problem because of Init() function, but could still zero initialize if this is not a hot class.